src/validation/coinbase.cairo

```diff
@@ -30,7 +32,43 @@ pub fn validate_coinbase(
     let block_reward = compute_block_reward(block_height);
     assert(total_output_amount <= total_fees + block_reward, 'total output > block rwd + fees');
 
-    // TODO: validate BIP-34 sig script
+    // validate BIP-34 sig script
+    if block_height >= BIP_34_BLOCK_HEIGHT {
+        let sig = *tx.inputs[0].script;
+        let sig_len = sig.len();
+
+        // Ensure byte length greater than 2 and less 100
+        assert(sig_len > 2 && sig_len < 100, 'bad sig script length');
+
+        // Ensure script starts with the current block height
+        //
+        // First byte is number of bytes in the number (will be 0x03 on mainnet for the next
+        // 150 or so years with 223-1 blocks), following bytes are little-endian representation
+        // of the number
+        //
+        // calculate block height from little-endian representation
+        let number_of_bytes = sig[0];
```

Comment:
@od-hunter you don't have to implement decoding, everything was fine, you just need to check that the first byte is indeed `0x03`. If there's an invalid/malicious script we have to catch that

Reply:
@m-kus Okay, working on it now